### PR TITLE
[cli] add command to set mesh-local prefix

### PR DESCRIPTION
**Pull request by jwhui**
Also add an expect test for getting/setting mesh-local prefix.

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -3564,7 +3564,18 @@ template <> otError Interpreter::Process<Cmd("prefix")>(Arg aArgs[])
...
+        if (aArgs[1].IsEmpty())
+        {
+            OutputIp6PrefixLine(*otThreadGetMeshLocalPrefix(GetInstancePtr()));
+        }
+        else
+        {
+            otIp6Prefix prefix;
+
+            SuccessOrExit(error = aArgs[1].ParseAsIp6Prefix(prefix));
+            error =
```
Comment (abtink) [inline src/cli/cli.cpp:3576]: Should we add a check that for prefix length being 64?

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: tests/scripts/expect/cli-misc.exp [hunk 2]
```diff
@@ -180,6 +180,11 @@ expect "Done"
 send "ba state\n"
 expect "Done"
 
+send "prefix meshlocal fd00:dead:beef:cafe::/64\n"
+expect_line "Done"
```
Comment (bukepo) [inline tests/scripts/expect/cli-misc.exp:184]:
```suggestion
send "prefix meshlocal fd00:dead:beef:cafe::/96\n"
expect_line "Error 7: InvalidArgs"
send "prefix meshlocal fd00:dead:beef:cafe::/64\n"
expect_line "Done"
```

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Pull request merged**